language/move-compiler/src/to_bytecode/translate.rs

```diff
@@ -465,13 +443,31 @@ fn var_info(
 // Structs
 //**************************************************************************************************
 
+fn struct_defs(
+    context: &mut Context,
+    m: &ModuleIdent,
+    structs: UniqueMap<StructName, H::StructDefinition>,
+) -> Vec<IR::StructDefinition> {
+    let sorting: HashMap<Symbol, usize> = structs
+        .iter()
+        .map(|(_, sname, sdef)| (*sname, sdef.index))
+        .collect();
+    let mut structs_vec = structs
+        .into_iter()
+        .map(|(s, sdef)| struct_def(context, &m, s, sdef))
+        .collect::<Vec<_>>();
+    structs_vec.sort_by_key(|sdef| sorting[&sdef.value.name.0]);
```

Comment:
Could this and the sorting done for constants/functions below be written without needing this `sorting` table and indexing into it? i.e., something like this:

```rust
let mut sorted_structs = structs.into_iter().collect();
sorted_structs.sort_by_key(|(_, sdef)| sdef.index);
sorted_structs.into_iter().map(|(s, sdef)| struct_def(context, &m, s, sdef)).collect()
```

Reply:
ah, yeah that feels a bit better doesn't it. Thanks!